infrastructure/src/main/java/org/corfudb/infrastructure/BaseServer.java

```diff
@@ -180,6 +213,24 @@ private void doReset(CorfuMsg msg, ChannelHandlerContext ctx, IServerRouter r) {
         CorfuServer.restartServer(true);
     }
 
+    /**
+     * Reset the JVM. This mechanism leverages that corfu_server runs in a bash script
+     * which monitors the exit code of Corfu. If the exit code is 100, then it resets
+     * the server and DELETES ALL EXISTING DATA.
+     * @param req The incoming request message.
+     * @param ctx The channel context.
+     * @param r The server router.
+     */
+    @AnnotatedServerHandler(type = CorfuProtocol.MessageType.RESET)
+    private void handleReset(Request req, ChannelHandlerContext ctx, org.corfudb.infrastructure.protocol.IServerRouter r) {
+        log.warn("Remote reset requested from client {}", req.getHeader().getClientId());
+        //TODO(Zach): checkArgument(req.hasResetRequest());
```

Comment:
How can you even get routed to this point without having the ResetRequest? I think that check is done before a call is made to this method.

Reply:
I believe only the MessageType is checked. There hasn't been a check that a proper payload is provided.

Reply:
I think these checks can all be done in the handle method in RequestHandlerMethods.

Reply:
I decided to perform these checks in NettyServerRouter since it was originally performed in RequestHandler.